File: backendpy/models/trail_log.py
from sqlalchemy import Column, Integer, String, DateTime, Text
from sqlalchemy.ext.declarative import declarative_base
from datetime import datetime

Base = declarative_base()

class TrailLog(Base):
    __tablename__ = "trail_logs"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(String, index=True)
    action = Column(String, index=True)
    component = Column(String)
    details = Column(Text)  # JSON string
    timestamp = Column(DateTime, default=datetime.utcnow)
    user_agent = Column(String)
    ip_address = Column(String) 

File: backend/Models/TrailLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AthalaSIEM.Models
{
    public class TrailLog
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public string UserId { get; set; }
        
        [Required]
        public string Action { get; set; }
        
        [Required]
        public string Component { get; set; }
        
        [Column(TypeName = "jsonb")]
        public string Details { get; set; }
        
        [Required]
        public DateTime Timestamp { get; set; }
        
        [Required]
        public string UserAgent { get; set; }
        
        public string IpAddress { get; set; }
        
        // Navigation property
        public virtual ApplicationUser User { get; set; }
    }
} 